app/components/trainees/record_details/view.rb

```diff
@@ -25,6 +25,38 @@ def last_updated_date
         render_text_with_hint(trainee.updated_at)
       end
 
+      def trainee_rows
```

Comment:
Code style comment, but it might read a little easier if you pull the logic for choosing what trn row to render into its own method. Then you don't have to do any array `<<` appending - just call the method in place:

```
def trainee_rows
[
    {
      key: "Trainee ID",
      value: trainee_id,
      action: govuk_link_to('Change<span class="govuk-visually-hidden"> trainee ID</span>'.html_safe,	edit_trainee_trainee_id_path(trainee)),
    },
    trn_row,
    {
      key: "Last updated",
      value: last_updated_date,
    },
    {
      key: "Record created",
      value: date_for_summary_view(trainee.created_at),
    },
  ]
end

Reply:
Good idea!